packages/payment-detection/test/erc20/fee-proxy-contract.test.ts

```diff
@@ -0,0 +1,158 @@
+    erc20FeeProxyContract = new ERC20FeeProxyContract({ advancedLogic: mockAdvancedLogic });
+  });
+
+  it('can createExtensionsDataForCreation', async () => {
+    const spy = sandbox.on(
+      mockAdvancedLogic.extensions.feeProxyContractErc20,
+      'createCreationAction',
+    );
+
+    await erc20FeeProxyContract.createExtensionsDataForCreation({
+      paymentAddress: 'ethereum address',
+      salt: 'ea3bc7caf64110ca',
+    });
+
+    expect(spy).to.have.been.called.with({
+      feeAddress: undefined,
+      feeAmount: undefined,
+      paymentAddress: 'ethereum address',
+      refundAddress: undefined,
+      salt: 'ea3bc7caf64110ca',
+    });
+  });
+
+  it('can createExtensionsDataForCreation with fee amount and address', async () => {
+    const spy = sandbox.on(
+      mockAdvancedLogic.extensions.feeProxyContractErc20,
+      'createCreationAction',
+    );
+
+    await erc20FeeProxyContract.createExtensionsDataForCreation({
+      feeAddress: 'fee address',
+      feeAmount: '2000',
+      paymentAddress: 'ethereum address',
+      salt: 'ea3bc7caf64110ca',
+    });
+
+    expect(spy).to.have.been.called.with({
+      feeAddress: 'fee address',
+      feeAmount: '2000',
+      paymentAddress: 'ethereum address',
+      refundAddress: undefined,
+      salt: 'ea3bc7caf64110ca',
+    });
+  });
+
+  it('can createExtensionsDataForCreation without salt', async () => {
+    const spy = sandbox.on(
+      mockAdvancedLogic.extensions.feeProxyContractErc20,
+      'createCreationAction',
+    );
+
+    await erc20FeeProxyContract.createExtensionsDataForCreation({
+      paymentAddress: 'ethereum address',
+    });
+
+    // Can't check parameters since salt is generated in createExtensionsDataForCreation
+    expect(spy).to.have.been.called;
+  });
+
+  it('can createExtensionsDataForAddPaymentInformation', async () => {
+    const spy = sandbox.on(
+      mockAdvancedLogic.extensions.feeProxyContractErc20,
+      'createAddPaymentAddressAction',
+    );
+
+    erc20FeeProxyContract.createExtensionsDataForAddPaymentInformation({
+      paymentAddress: 'ethereum address',
+    });
+
+    expect(spy).to.have.been.called.with({
+      paymentAddress: 'ethereum address',
+    });
+  });
+
+  it('can createExtensionsDataForAddRefundInformation', async () => {
+    const spy = sandbox.on(
+      mockAdvancedLogic.extensions.feeProxyContractErc20,
+      'createAddRefundAddressAction',
+    );
+
+    erc20FeeProxyContract.createExtensionsDataForAddRefundInformation({
+      refundAddress: 'ethereum address',
+    });
+
+    expect(spy).to.have.been.called.with({
+      refundAddress: 'ethereum address',
+    });
+  });
+
+  it('can createExtensionsDataForAddFeeInformation', async () => {
+    const spy = sandbox.on(
+      mockAdvancedLogic.extensions.feeProxyContractErc20,
+      'createAddFeeAction',
+    );
+
+    erc20FeeProxyContract.createExtensionsDataForAddFeeInformation({
+      feeAddress: 'ethereum address',
```

Comment:
at which step do we validate the feeAddress?

Reply:
we will do it when we calculate the feeBalance